static/js/util.js

```diff
@@ -254,6 +254,19 @@ exports.prefix_sort = function (query, objs, get_item) {
              rest:    noMatch };
 };
 
+// manipulate prefix_sort to select popular emojis first
+exports.emoji_prefix_sort = function (query, objs, get_item) {
+    var prefix_sort = exports.prefix_sort(query, objs, get_item);
+    var popular_emojis = [];
+    var sorted_emojis = [];
+    prefix_sort.matches.forEach(function (obj) {
+        var list_select = emoji_picker.frequently_used_emojis_list.includes(obj.codepoint) ?
+                          popular_emojis : sorted_emojis;
+        list_select.push(obj);
```

Comment:
This is a really hard to read way to write this.  It would be much more readable as:
```
+        if (emoji_picker.frequently_used_emojis_list.includes(obj.codepoint)) {
+            popular_emoji_matches.push(obj);
+        } else {
+            other_emoji_matches.push(obj);
+        }
```

(Also renamed the variables a bit).  

It's a bit more lines of code, but it's much easier to understand at a glance what it does.

In a large project, ease of reading is far more important than compactness, since code gets read at least 10x as many times as it gets written :).

Reply:
Sure yeah I'll remember that :)